### PR TITLE
[AT-1455] add an example for auto closable/closable TimestreamWriteClient

### PR DESCRIPTION
**Pull request by dkovalenkoI**
## Summary
Suggestion - For the Java examples, show that the ***TimeStreamWriteClient*** is closable

## Description
The problem related to creating ***TimeStreamWriteClient*** in case of error. New clients send initialization error due too previous client was not closed.

In the example logic of using client has been modified to show the ability of ***TimestreamWriteClient*** to be used in autoclosable condition. 
In the example we create new client and use it in try catch block to show how it the client can be closed.

## Related Issue
[awslabs#99](https://github.com/awslabs/amazon-timestream-tools/issues/99) 

## Tests performed/created
Following Tests has been executed manually:

### Precondition:
- Build the project: ***mvn clean install***
- Run the project using command: ***mvn install exec:java -Dexec.mainClass="com.amazonaws.samples.kinesis2timestream.StreamingJob" -Dexec.args="--InputStreamName TimestreamTestStream --Region us-east-1 --TimestreamDbName kdaflink --TimestreamTableName kinesisdata" -Dexec.classpathScope=test***

### Results
Timestream database ***kdaflink*** with table ***kinesisdata*** in region ***us-east-1*** has been created

## Additional Reviewers
@alexeytemnikov

### CODE REVIEW

File: integrations/flink_connector/sample-kinesis-to-timestream-app/src/main/java/com/amazonaws/samples/kinesis2timestream/TimestreamInitializer.java [hunk 1]
```diff
@@ -15,31 +15,23 @@
  * Checks if required database and table exists in Timestream. If they do not exists, it creates them
  */
 public class TimestreamInitializer {
-    private final TimestreamWriteClient writeClient;
```
Comment (RoyZhang2022) [inline integrations/flink_connector/sample-kinesis-to-timestream-app/src/main/java/com/amazonaws/samples/kinesis2timestream/TimestreamInitializer.java:18]: Just wondering for user case the TimestreamInitializer object is not destroyed? If not destroyed, the writeClient is not closed. If it is destroyed, writeClient should be closed in the destruction phase.

**Comment by alexey-temnikov** [comment]
Use the following PR Template:

## PR Template

### Summary


### Description


### Related Issue


### Tests performed/created


### Additional Reviewers
@alexeytemnikov

File: integrations/flink_connector/sample-kinesis-to-timestream-app/src/main/java/com/amazonaws/samples/kinesis2timestream/TimestreamInitializer.java [hunk 2]
```diff
@@ -74,4 +67,21 @@ private URI parseEndpointOverride(String endpointOverride) {
             throw new RuntimeException("Invalid EndpointOverride Config: " + endpointOverride);
         }
     }
+
+    private TimestreamWriteClient createClient() {
```
Comment (alexey-temnikov) [inline integrations/flink_connector/sample-kinesis-to-timestream-app/src/main/java/com/amazonaws/samples/kinesis2timestream/TimestreamInitializer.java:71]: final?
```suggestion
    private final TimestreamWriteClient createClient() {
```

File: integrations/flink_connector/sample-kinesis-to-timestream-app/src/main/java/com/amazonaws/samples/kinesis2timestream/TimestreamInitializer.java [hunk 3]
```diff
@@ -15,31 +15,23 @@
...
-            this.writeClient = TimestreamWriteClient
-                    .builder()
-                    .region(Region.of(region))
-                    .build();
-        }
+        this.region = region;
+        this.endpointOverride = endpointOverride;
     }
 
     public void createDatabase(String databaseName) {
```
Comment (alexey-temnikov) [inline integrations/flink_connector/sample-kinesis-to-timestream-app/src/main/java/com/amazonaws/samples/kinesis2timestream/TimestreamInitializer.java:26]: Would it be better to refactor it into 2 methods - createDatabase and createTable?

**Review by alexey-temnikov — DISMISSED** [review]
Dima, please addess copmments above.

File: integrations/flink_connector/sample-kinesis-to-timestream-app/src/main/java/com/amazonaws/samples/kinesis2timestream/TimestreamInitializer.java (hunk 1 above)
Comment (dkovalenkoI) [reply integrations/flink_connector/sample-kinesis-to-timestream-app/src/main/java/com/amazonaws/samples/kinesis2timestream/TimestreamInitializer.java:18]: @RoyZhang2022 , in java we don't need to destroy objects. In the example we use try(TimestreamWriteClient client = createClient()) that equal to 
TimestreamWriteClient client = createClient()
try() {
//some logic
} finally {
client.close();
}

**Comment by RoyZhang2022** [comment]
For the description part, "Related Issue" should be the link to the Jira.

**Review by RoyZhang2022 — APPROVED** [review]

**Review by alexey-temnikov — APPROVED** [review]

**Pull request merged**